File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String s1 = strs[0], s2 = strs[strs.length - 1];
        int i = 0;
        short n1 = (short)s1.length(), n2 = (short)s2.length();
        
        while (i < n1 && i < n2) {
            if (s1.charAt(i) != s2.charAt(i)) {
                break;
            }
            i++;
        }
        
        return s1.substring(0, i);
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        
        string s = strs[0], t = strs[strs.size() - 1];
        
        string res;
        
        for (int i = 0, n = min(s.size(), t.size()); i < n; i++) {
            if (s[i] != t[i]) break;
            res = res + s[i];
        }
        
        return res;
    }
};